app/components/date_single/edit_component.html.erb

```diff
@@ -0,0 +1,29 @@
+<%= tag.div data: { controller: 'form-date' }, class: reset_class  do %>
+  <div class="row">
+    <div class="col-3">
+      <%= render Elements::Forms::TextFieldComponent.new(form:, field_name: :year, label: helpers.t('date.fields.year.label'),
+                                                         data: { form_date_target: 'year', action: 'form-date#update' }, container_classes: 'mb-2') %>
+    </div>
+    <div class="col-5">
+      <%= render Elements::Forms::SelectMonthFieldComponent.new(form:, field_name: :month, label: helpers.t('date.fields.month.label'),
+                                                                data: { form_date_target: 'month', action: 'form-date#update' }) %>
+    </div>
+    <div class="col-3">
+      <%= render Elements::Forms::SelectDayFieldComponent.new(form:, field_name: :day, label: helpers.t('date.fields.day.label'),
+                                                              data: { form_date_target: 'day', action: 'form-date#update' }) %>
+    </div>
+    <% if with_reset? %>
+      <%= tag.div class: 'col-1 d-flex align-items-stretch mt-3', data: { controller: 'form-date-reset', form_date_reset_form_date_outlet: ".#{reset_class}" } do %>
```

Comment:
Using stimulus outlets to allow a single reset button to reset both dates in a date range.